book/src/concepts/system.md

```diff
@@ -290,6 +290,153 @@ write_storage.remove(entity);
 
 Keep in mind that inserting a component on an entity that already has a component of the same type **will overwrite the previous one**.
 
+## Changing states through resources
+
+In a previous section we talked about [`States`][s], and how they are used to organize your game
+into different logical sections.
+Since systems are responsible for processing user input, they must somehow also be responsible for
+causing a switch in states.
```

Comment:
pssst, that isn't exactly true. The best way to do it is to have systems only to data processing (change components, modify resources, render things). When you have systems that manage states, they get too specialized and less re-usable. 

The best practice for those case would be to emit an event specific to your game from the System. For example, a "TogglePause" event. Then, inside of your state's handle_event method, you can react to this event (in this case, by returning a Trans when you see the "TogglePause" event).

Reply:
> pssst, that isn't exactly true. The best way to do it is to have systems only to data processing (change components, modify resources, render things). When you have systems that manage states, they get too specialized and less re-usable.
> 
> The best practice for those case would be to emit an event specific to your game from the System. For example, a "TogglePause" event. Then, inside of your state's handle_event method, you can react to this event (in this case, by returning a Trans when you see the "TogglePause" event).

To me it seems like a system still seems to be responsible for the state change? How would you like the phrasing to change?

Reply:
This is an interesting discussion. In any case, we can currently process input in both location, so I think this statement can be confusing.

Reply:
I added some clarification with an example. Please check it out!